rest-api/src/controllers/ImageController.ts

```diff
@@ -7,411 +13,471 @@ import { ImageService } from "../client/images/ImageService";
+      ctx.status = 204;
+    } catch (e) {
+      if (e instanceof LikeImageError) {
+        switch (e.reason) {
+          case LikeImageResponseStatus.IMAGE_NOT_FOUND_ERROR:
+            throw new RequestError(404, "Image not found");
+          case LikeImageResponseStatus.USER_NOT_FOUND_ERROR:
+            throw new RequestError(404, "User not found");
+          default:
+            ctx.log.warn(e);
+            throw new RequestError(500);
         }
+      }
+      ctx.log.error(e);
+      throw new RequestError(503);
     }
+  };
 
-    /**
-     * Like about an image.
-     */
-    private likeImage = async (
-        ctx: RouterContext<AuthState & IValidationState>
-    ) => {
-        const results = validationResults(ctx as any); // as any because of bad typings...
-        if (results.hasErrors()) {
-            throw new RequestError(422, { errors: results.array() });
+  /**
+   * Get image likes.
+   */
+  private getLikes = async (ctx: RouterContext<IValidationState>) => {
+    const results = validationResults(ctx);
+    if (results.hasErrors()) {
+      throw new RequestError(422, { errors: results.array() });
+    }
+    const { imageId, page, size } = results.passedData();
+    const parsedPage = page ? parseInt(page, 10) : 1;
+    const parsedSize = size ? parseInt(size, 10) : 10;
+    try {
+      const likes = await this.imageService.getLikes(
+        imageId,
+        parsedPage,
+        parsedSize
+      );
+      ctx.body = likes;
+    } catch (e) {
+      if (e instanceof GetLikesError) {
+        switch (e.reason) {
+          case GetLikesErrorStatus.GET_LIKES_IMAGE_NOT_FOUND:
+            throw new RequestError(404, "Image not found");
+          default:
+            ctx.log.warn(e);
+            throw new RequestError(500);
         }
+      }
+      ctx.log.error(e);
+      throw new RequestError(503);
+    }
+  };
 
-        const { imageId, unlike: ulike } = results.passedData();
-        const unlike = typeof ulike === "boolean" ? ulike : false;
-        const { id } = ctx.state.account;
-
-        try {
-            await this.imageService.likeImage(imageId, id, unlike);
-            ctx.status = 204;
-        } catch (e) {
-            if (e instanceof LikeImageError) {
-                switch (e.reason) {
-                    case LikeImageResponseStatus.IMAGE_NOT_FOUND_ERROR:
-                        throw new RequestError(404, "Image not found");
-                    case LikeImageResponseStatus.USER_NOT_FOUND_ERROR:
-                        throw new RequestError(404, "User not found");
-                    default:
-                        ctx.log.warn(e);
-                        throw new RequestError(500);
-                }
-            }
-            ctx.log.error(e);
-            throw new RequestError(503);
+  private postComment = async (
+      ctx: Router.RouterContext<AuthState & IValidationState>) => {
+    const results = validationResults(ctx as any);
```

Comment:
Checking the validation results is pointless if there are no validations declared for the route